ctsit/nacculator
# Make Z1 and C1S forms optional, update filters, and update documentation

This change adds the ability for the UDS3 `builder.py` files to skip building the Z1 and C1S forms if they are not present in an ADRC's REDCap project and/or exported csv. 
This branch also includes a revision to the README file that cleans up the formatting and clarifies terms for general usage. The `run_filters.py` file has been moved into the nacc/ folder so that it can be installed as a custom command (so that nacculator now uses `redcap2nacc` and `nacculator_filters`), and the config file (`nacculator_cfg.ini.example`) has been modified to include documentation on the various filters, and the ability to switch them off individually and customize them a little further.

This change was made to address issues [108](https://github.com/ctsit/nacculator/issues/108) and [109](https://github.com/ctsit/nacculator/issues/109).


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. From the command-line, run:
`$ git clone https://github.com/ctsit/nacculator.git nacculator`
`$ cd nacculator`
`$ python3 -mvenv venv`
`$ source venv/bin/activate`
`$ pip install -e .`
*(note: for this change to be visible, you must be using the 'feature/make_forms_optional' branch in the nacculator clone)*
2. Then, fill out the `nacculator_cfg.ini.example` file in the working directory and save as `nacculator_cfg.ini`
3. Run the commands:
`$ nacculator_filters path/to/nacculator_cfg.ini`
`$ redcap2nacc -ivp < run_$(CURRENT-DATE)/final_Update.csv > data_ivp.txt`
`$ redcap2nacc -fvp < run_$(CURRENT-DATE)/final_Update.csv > data_fvp.txt`
4. The output should look like the filters being run (with progress being printed to the command line), and then two regular nacculator output .txt files that are immediately ready to submit to NACC, with errors printed to the command line (stderr).
5. If you pass data with errors, the output should look like regular errors being caught by nacculator's error checkers, with those particular PTIDs and visits skipped.
6. You have successfully installed nacculator and run two basic commands, with or without the Z1 form present in your REDCap project.

- Verify the thing does this: properly processes raw data immediately (after creating the config file) on a fresh install.
- Verify the thing does not do that: return improper errors or refuse to run.


## Affirmations
